schematic/schemas/data_model_graph.py

```diff
@@ -226,6 +229,64 @@ def get_adjacent_nodes_by_relationship(
 
         return list(nodes)
 
+    def get_component_node_validation_rules(
+        self, manifest_component: str, node_display_name: str
+    ) -> list[str]:
+        """Get valdation rules for a given node and component.
+        Args:
+            manifest_component, str: manifest component display name that the node belongs to.
+            node_display_name, str: node display name for the node being queried.
+        Returns:
+            validation_rules, list[str]: validation rules list for a given node and component.
+        """
+        # get any additional validation rules associated with this node (e.g. can this node be mapped to a list of other nodes)
+        node_validation_rules = self.get_node_validation_rules(
+            node_display_name=node_display_name
+        )
+
+        # Parse the validation rules per component if applicable
+        if node_validation_rules and isinstance(node_validation_rules, dict):
+            node_validation_rules = extract_component_validation_rules(
+                manifest_component=manifest_component,
+                validation_rules_dict=node_validation_rules,
+            )
+        return node_validation_rules
+
+    def get_component_node_required(
+        self, manifest_component, node_display_name
+    ) -> bool:
+        """Check if a node is required taking into account the manifest component it is defined in (requirements can be set in validaiton rule as well as required column)
+        Args:
+            manifest_component, str: manifest component display name that the node belongs to.
+            node_display_name, str: node display name for the node being queried.
+        Returns:
+            True, if node is required, False if not
+        """
+        node_required = False
+        # Get node validation rules for a given component
+        node_validation_rules = self.get_component_node_validation_rules(
+            manifest_component=manifest_component, node_display_name=node_display_name
+        )
+
+        # Check if the valdation rule specifies that the node is required for this particular component.
+        if rule_in_rule_list("required", node_validation_rules):
+            node_required = True
+            # To prevent any unintended errors, ensure the Required field for this node is False
+            if (
+                self.get_node_required(node_display_name=node_display_name)
+                and node_required
```

Comment:
I don't think `and node_required` is needed here as it will always be true by it being se to true at the beginging of this if block

Reply:
Good call!